tabular/src/autogluon/tabular/features/generators/drop_unique.py

```diff
@@ -36,10 +36,16 @@ def _drop_unique_features(X: DataFrame, feature_metadata: FeatureMetadata, max_u
         max_unique_value_count = X_len * max_unique_ratio
         for column in X:
             unique_value_count = len(X[column].unique())
+            # Drop features that are always the same
             if unique_value_count == 1:
                 features_to_drop.append(column)
-            elif feature_metadata.get_feature_type_raw(column) in [R_CATEGORY, R_OBJECT] and (unique_value_count > max_unique_value_count):
-                features_to_drop.append(column)
+            elif feature_metadata.get_feature_type_raw(column) in [R_CATEGORY, R_OBJECT]\
```

Comment:
Please also split this change out to the same PR as mentioned in the above comment.